react-toggle-switch-component/app-name/src/ToggleSwitch.tsx

```diff
@@ -0,0 +1,25 @@
+import { useState } from 'react';
+
+export function ToggleSwitch() {
+  const [isOn, setIsOn] = useState(false);
+
+  function handleToggleOn() {
+    setIsOn(() => !isOn);
```

Comment:
You don't need the function form of the setter here. Just use `setIsOn(!isOn);`. There are actually very few times that you need the function form of state setters. I use that form about one time out of every 100 times I use the non-function form, maybe less!